source/Halibut/Transport/Protocol/MessageExchangeProtocol.cs

```diff
@@ -40,10 +43,19 @@ public ResponseMessage ExchangeAsClient(RequestMessage request)
 
         public async Task<ResponseMessage> ExchangeAsClientAsync(RequestMessage request, CancellationToken cancellationToken)
         {
-            await PrepareExchangeAsClientAsync(cancellationToken);
+            try
```

Comment:
This is where it gets called when we are giving a request to a listening tentacle